src/site/content/en/blog/sw-range-requests/index.md

```diff
@@ -0,0 +1,95 @@
+## What's the issue?
+
+Consider a service worker with the following `fetch` event listener, which takes every incoming request and passes it to the network:
+
+```javascript
+self.addEventListener('fetch', (event) => {
+  // The Range: header will not pass through in
+  // browsers that behave incorrectly.
+  event.respondWith(fetch(event.request));
+});
+```
+
+{% Aside %}
+This sort of trivial `fetch` event listener should [normally be avoided](https://developers.google.com/web/fundamentals/primers/service-workers/high-performance-loading#never_use_a_passthrough_fetch_handler); it's used here for illustrative purposes.
+{% endAside %}
+
+In browsers with the incorrect behavior, if `event.request` included a `Range:` header, that header would be silently dropped. The request that was received by the remote server would not include `Range:` at all. This would not necessarily "break" anything, since a server is _technically_ allowed to return the full response body, with a <code>[200 status code](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/200)</code>, even when a <code>Range:</code> header is present in the original request. But it would result in more data being transferred than is strictly needed from the perspective of the browser.
+
+Developers who were aware of this behavior could work around it by explicitly checking for the presence of a `Range:` header, and not calling `event.respondWith()` if one is present. By doing this, the service worker effectively removes itself from the response generation picture, and the default browser networking logic, which knows how to preserve range requests, is used instead.
+
+```javascript
+self.addEventListener('fetch', (event) => {
+  // Return without calling event.respondWith()
+  // if this is a range request.
+  if (event.request.headers.has('range')) {
+    return;
+  }
+
+  event.respondWith(fetch(event.request));
+});
+```
+
+It's safe to say that most developers were not aware of the need to do this, though. And it wasn't clear _why_ that should be required. Ultimately, this limitation was due to [browsers](https://bugs.chromium.org/p/chromium/issues/detail?id=847428) needing to catch up to [changes in the underlying specification](https://github.com/whatwg/fetch/pull/560), which added support for this functionality.
+
+## What's been fixed?
+
+Browsers that behave correctly preserve the `Range:` header when `event.request` is passed to `fetch()`. This means the service worker code in my initial example will allow the remote server to see the `Range:` header, if it was set by the browser:
+
+```javascript
+self.addEventListener('fetch', (event) => {
+  // The Range: header will pass through in browsers
+  // that behave correctly.
+  event.respondWith(fetch(event.request));
+});
+```
+
+The server now gets a chance to properly handle the range request and return a partial response with a <code>[206 status code](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/206)</code>.
```

Comment:
What's the reasoning for making `206 status code` code-formatted? Seems like this should be regular text.

Reply:
That was an artifact produced by the Google Doc to Markdown conversion script. I'll fix here and elsewhere.